### PR TITLE
goal: allow ConfigJSONOverride in local network templates

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Allow ConfigJSONOverride in `goal network create` templates similarly to `netgoal`.

## Test Plan

Tested manually

### CODE REVIEW

File: test/testdata/nettemplates/perf/mkconf.py [hunk 1]
```diff
@@ -33,10 +34,20 @@
 
   d["Nodes"].append(node)
 
+npn_nodes = 50
+for n in range(0, npn_nodes):
+  node = {
+    "Name": "NPNode%d" % n,
+    "Wallets": [],
+    "DeadlockDetection": -1,
+    "ConfigJSONOverride": '{"ForceFetchTransactions":true}'
```
Comment (brianolson) [inline test/testdata/nettemplates/perf/mkconf.py:43]: This is what the change is really about. So, why this?

**Review by brianolson — COMMENTED** [review]
Go code looks fine, I wrote a unit test that I sent a PR about to add to this.
What's the implication of the mkconf.py change?

File: test/testdata/nettemplates/perf/mkconf.py (hunk 1 above)
Comment (algorandskiy) [reply test/testdata/nettemplates/perf/mkconf.py:43]: I used mkconf to generate local topologies for some tests, and decided to leave ConfigJSONOverride example there. I can remove if it does not make much sense having it here.

**Comment by winder** [comment]
Nice. This is semi related to a "profile" feature we've been working on https://github.com/algorand/go-algorand/pull/5069

**Comment by brianolson** [comment]
I think the mkconf.py change is a little weird as an "example" because it has concrete consequences that aren't necessarily intended?
How about this additional test to ensure that `encoding/json` does overlay behavior correctly? https://github.com/algorandskiy/go-algorand/pull/6

**Comment by algorandskiy** [comment]
fair, thank you, applied

**Review by brianolson — DISMISSED** [review]

File: netdeploy/networkTemplates_test.go [hunk 2]
```diff
@@ -95,3 +97,21 @@ func TestValidate(t *testing.T) {
...
 	a.NoError(err)
 }
+
+type overlayTestStruct struct {
+	A string
+	B string
+}
+
+// TestJsonOverlay ensures that encoding/json will only clobber fields present in the json and leave other fields unchanged
+func TestJsonOverlay(t *testing.T) {
```
Comment (algonautshant) [inline netdeploy/networkTemplates_test.go:107]:
```suggestion
func TestJsonOverlay(t *testing.T) {
	partitiontest.PartitionTest(t)
```

**Review by winder — APPROVED** [review]

File: netdeploy/networkTemplates_test.go [hunk 3]
```diff
@@ -95,3 +97,22 @@ func TestValidate(t *testing.T) {
...
```
Comment (algonautshant) [inline netdeploy/networkTemplates_test.go:107]: What is the purpose of this test? It looks like it is a json and dec test, and not a netdeploy test.
Ideally, the test should test `netdeploy/`.

**Review by algonautshant — APPROVED** [review]
Looks fine. Very useful.

**Pull request merged**